core/workspace_comment_render_svg.js

```diff
@@ -276,40 +368,46 @@ Blockly.WorkspaceCommentSvg.prototype.resizeMouseDown_ = function(e) {
   e.stopPropagation();
 };
 
-/**
- * Handle a mouse-down on comment's delete icon.
- * @param {!Event} e Mouse down event.
- * @private
- */
-Blockly.WorkspaceCommentSvg.prototype.deleteMouseDown_ = function(e) {
-  // highlight the delete icon
-  Blockly.utils.addClass(
-      /** @type {!Element} */ (this.deleteIconBorder_), 'blocklyDeleteIconHighlighted');
-  // This event has been handled.  No need to bubble up to the document.
-  e.stopPropagation();
-};
 
 /**
- * Handle a mouse-out on comment's delete icon.
- * @param {!Event} e Mouse out event.
+ * Set the minimized state of the bubble.
+ * @param {boolean} minimize Whether the bubble should be minimized
+ * @param {?string} labelText Optional label text for the comment top bar
+ *    when it is minimized.
  * @private
  */
-Blockly.WorkspaceCommentSvg.prototype.deleteMouseOut_ = function(/*e*/) {
-  // restore highlight on the delete icon
-  Blockly.utils.removeClass(
-      /** @type {!Element} */ (this.deleteIconBorder_), 'blocklyDeleteIconHighlighted');
-};
-
-/**
- * Handle a mouse-up on comment's delete icon.
- * @param {!Event} e Mouse up event.
- * @private
- */
-Blockly.WorkspaceCommentSvg.prototype.deleteMouseUp_ = function(e) {
-  // Delete this comment
-  this.dispose(true, true);
-  // This event has been handled.  No need to bubble up to the document.
-  e.stopPropagation();
+Blockly.WorkspaceCommentSvg.prototype.setRenderedMinimizeState_ = function(minimize, labelText) {
```

Comment:
You have this function and another one (`setMinimized`) that calls it and does some related things. What's the relationship between them?  When would this one be called without calling the other?

Some of the explanation should go into the comments as well.

Reply:
`setRenderedMinimizeState` gets called from `setMinimized` *iff* this comment is rendered. `setMinimized` is responsible for keeping track of the minimize state regardless of whether or not this comment is rendered, and `setRenderedMinimizeState` actually changes the rendering to the minimized form (e.g. just the top bar shows, the truncated comment text in the top bar, etc.).  In block comment land, these happened in two separate classes. The equivalent of `setMinimized` happened in `Blockly.ScratchBlockComment` and the equivalent of `setRenderedMinimizeState` happened in `Blockly.ScratchBubble`.

I will add some of this explanation to the comments! =D

Reply:
Done.